src/reducers/searchReducer.js

```diff
@@ -9,6 +9,8 @@ const initialState = {
 
 export default function Search(state = initialState, action) {
   switch (action.type) {
+    case 'SET_TOOL':
+      return (action.tool == 'search') ? state : initialState;
```

Comment:
kludne si to zmen. ak si na to pri pisani spomeniem, tak sa pokusim v buducnosti pouzivat striktny operator.